### PR TITLE
add retry logic to client and add support for ruby 3 and single vector search

### PR DESCRIPTION
**Pull request by washu**
Add logic, all specs pass

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
 require "bundler/gem_tasks"
+require "bundler"
+Bundler.setup
+Bundler::GemHelper.install_tasks
+
+require "rake"
+require "yaml"
```
Comment (andreibondarev) [inline Rakefile:9]: What is requiring rake and yaml for?

File: lib/milvus/search.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ class Search < Base
     def post(
       collection_name:,
       anns_field:, top_k:, params:, metric_type:, vectors:, dsl_type:, output_fields: nil,
-      round_decimal: nil
+      round_decimal: nil, partition_names: nil,vector: nil, filter: nil
```
Comment (andreibondarev) [inline lib/milvus/search.rb:10]: @washu Could you please send me a link to the documentation where you found that these params are supported?

Comment (washu) [reply lib/milvus/search.rb]: you are correct i mis read a GRPC call for http api call. reverted this out

File: Rakefile (hunk 1 above)
Comment (washu) [reply Rakefile]: fixed

File: milvus.gemspec [hunk 3]
```diff
@@ -12,7 +12,7 @@ Gem::Specification.new do |spec|
   spec.description = "Ruby wrapper for the Milvus vector search database API"
   spec.homepage = "https://github.com/andreibondarev/milvus"
   spec.license = "MIT"
-  spec.required_ruby_version = ">= 2.6.0"
+  spec.required_ruby_version = ">= 3.0"
```
[the remark this answers is not in the document]
Comment (washu) [reply milvus.gemspec:15]: that would make sense, 2.7 is EOL already, 3 will be in 8 months.

**Comment by washu** [comment]
also it brings to milvus 2 i guess compat. thats the version im using

**Comment by andreibondarev** [comment]
@washu Any chance that we can separate this PR into separate ones? For example dropping Ruby 2.7 compatibility could be a separate PR.

**Comment by bendangelo** [comment]
Is there an update for this?

**Comment by andreibondarev** [comment]
@washu @bendangelo I switched the gem to use v2 API. If you're running Milvus yourself you should be using port: `19530`

**Pull request closed without merging**